app/routes/events/components/EventAttendeeStatistics.tsx

```diff
@@ -253,13 +259,36 @@ const Analytics = ({ eventId }: { eventId: ID }) => {
 
   const dispatch = useAppDispatch();
 
+  const memoizedData = useMemo(() => {
+    return async () => {
+      if (eventId) {
+        const response = await dispatch(fetchAnalytics(eventId));
+        let data = response.payload;
+
+        if (viewStartTime) {
+          data = data.filter(
+            (item) => new Date(item.date) >= new Date(viewStartTime as string)
+          );
+        }
+
+        if (viewEndTime) {
+          data = data.filter(
+            (item) => new Date(item.date) <= new Date(viewEndTime as string)
+          );
+        }
+        return data;
+      }
+    };
+  }, [eventId, viewStartTime, viewEndTime, dispatch]);
+
   useEffect(() => {
-    eventId &&
-      dispatch(fetchAnalytics(eventId)).then((res) => {
-        setData(res.payload);
-        setMetrics(calculateMetrics(res.payload));
+    if (eventId) {
+      memoizedData().then((data) => {
+        setData(data);
+        setMetrics(calculateMetrics(data));
```

Comment:
You can do the filtering here. Something like this. Feel free to change up the syntax and convention of this
```suggestion
      memoizedData().then((data) => {
        const filteredData = data.filter((item => {return /*Calculation here*/}))
        setData(filteredData);
        setMetrics(calculateMetrics(filteredData));
```